task/prefetch-dependencies/0.1/prefetch-dependencies.yaml

```diff
@@ -74,13 +79,164 @@ spec:
         yq 'del(.goproxy_url)' <<< "${CONFIG_FILE_CONTENT}" > /mnt/config/config.yaml
       fi
 
-  - image: quay.io/redhat-appstudio/cachi2:0.13.0@sha256:eb34cfe3fea20997eebd8164dc93eedb2fd7a60dc1fb4afcc1b1ff43df9d6667
+  - name: check-prefetch-input
+    image: quay.io/redhat-appstudio/cachi2:0.13.0@sha256:eb34cfe3fea20997eebd8164dc93eedb2fd7a60dc1fb4afcc1b1ff43df9d6667
     # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
     # the cluster will set imagePullPolicy to IfNotPresent
-    name: prefetch-dependencies
     env:
     - name: INPUT
       value: $(params.input)
+    script: |
+      if [ -z "${INPUT}" ]
+      then
+        # Confirm input was provided though it's likely the whole task would be skipped if it wasn't
+        echo "No prefetch will be performed because no input was provided for cachi2 fetch-deps"
+        echo "skip" > /shared/skip
+      fi
+
+  - name: register-red-hat
+    image: quay.io/redhat-appstudio/cachi2@sha256:eb34cfe3fea20997eebd8164dc93eedb2fd7a60dc1fb4afcc1b1ff43df9d6667
+    env:
+    - name: INPUT
+      value: $(params.input)
+    - name: ACTIVATION_KEY
+      value: $(params.ACTIVATION_KEY)
+    volumeMounts:
+    - mountPath: /activation-key
+      name: activation-key
+    results:
+    - name: registered
+      type: string
+    script: |
+      #!/bin/bash
+      if [ -f /shared/skip ]; then
+        echo "Skipping."
+        exit 0
+      fi
+
+      echo "false" > /shared/registered
+      ACTIVATION_KEY_PATH="/activation-key"
+
+      mkdir -p /shared/rhsm/entitlement
+      mkdir -p /shared/rhsm/consumer
+
+      if [ -e /activation-key/org ]; then
+        cp -r --preserve=mode "$ACTIVATION_KEY_PATH" /tmp/activation-key
+
+        echo "Registering with Red Hat subscription manager."
+        subscription-manager register --org "$(cat /tmp/activation-key/org)" --activationkey "$(cat /tmp/activation-key/activationkey)"
+
+        # copy generated certificates to /shared/rhsm
+        cp /etc/pki/entitlement/*.pem /shared/rhsm/entitlement/
```

Comment:
Related to my earlier comment on not splitting this among different steps, I'm confused why we'd want a shared volume full of these default pre-populated subman system locations with data instead of handling this within the same context and referencing only these standard and expected system locations instead of shared volumes.

Reply:
I suspect that this bit of code just evolved this way but I tried to keep it consistent between the buildah task and this one. If we want to try to remove the copying and just use the volume directly I would prefer to handle it in a followup PR in both this task _and_ the buildah task. Side note: this could be a candidate for use with future stepActions.

Reply:
> I suspect that this bit of code just evolved this way but I tried to keep it consistent between the buildah task and this one.

Is this: https://github.com/konflux-ci/build-definitions/blob/main/task/buildah-oci-ta/0.2/buildah-oci-ta.yaml#L436-L462 the bit of buildah task work you're referring to?

Reply:
the general pattern even predates that section - [this bit](https://github.com/konflux-ci/build-definitions/blob/bf3a7f2c00c3863ca68566942d1a40c43175bcb5/task/buildah-oci-ta/0.2/buildah-oci-ta.yaml#L470) existed before activation keys were supported at all. There was some issue with trying to mount directly but I cannot recall what it was.

I also think that this code can be consolidated into a [stepAction](https://tekton.dev/vault/pipelines-v0.53.x-lts/stepactions/) in the future and to do that we 1) need to keep the data on a shared emptyDir as it is now and 2) would benefit from keeping the code in sync as best we can between prefetch and buildah tasks.